src/IHaskell/Types.hs

```diff
@@ -56,25 +65,32 @@ class IHaskellDisplay a where
 
 -- | Display as an interactive widget.
 class IHaskellDisplay a => IHaskellWidget a where
-  -- | Output target name for this widget. The actual input parameter should be ignored.
+  -- | Output target name for this widget. The actual input parameter should be ignored. By default
+  -- evaluate to "ipython.widget", which is used by IPython for its backbone widgets.
   targetName :: a -> String
+  targetName _ = "ipython.widget"
```

Comment:
I read up on `Data.Proxy` and finally understood what it was. Then I tried to use it, and it seems that it won't work here. It causes problems with defining an `IHaskellWidget` instance for `Widget`.

```haskell
targetName (Widget w) = targetName w -- Current definition
targetName (Proxy :: Widget) = targetName (Proxy :: ???)
```